controllers/helmrelease_controller_chart.go

```diff
@@ -47,6 +48,11 @@ func (r *HelmReleaseReconciler) reconcileChart(ctx context.Context, hr *v2.HelmR
 		Name:      hr.GetHelmChartName(),
 	}
 
+	if r.NoCrossNamespaceRef && chartName.Namespace != hr.Namespace {
```

Comment:
I don't think this is the right place for this, the chart shouldn't be created if the source is in a different namespace. We need to check the `sourceRef.namespace` way before it reaches this code.

Reply:
The chart is created after below line.
https://github.com/fluxcd/helm-controller/blob/main/controllers/helmrelease_controller_chart.go#L66

Reply:
Ok, we could tell users about the thing they've set in the spec which is the source (HelmRepo/GitRepo/Bucket). Instead of the `can't access 'HelmChart/chart-ns/chart-name'` could be `HelmRepository/repo-ns/repo-name`.